src/Z3/Base.hs

```diff
@@ -811,27 +821,14 @@ mkFiniteDomainSort = liftFun2 z3_mk_finite_domain_sort
 mkArraySort :: Context -> Sort -> Sort -> IO Sort
 mkArraySort = liftFun2 z3_mk_array_sort
 
-{- TODO
-data TupleTyple
-  = TupleType {
-      tupleSort :: Sort
-    , tupleCons :: FunDecl
-    , tupleProj :: [FunDecl]
-    }
-
-mkTupleSort :: ... -> IO TupleType
--}
-
 -- | Create a tuple type
 --
 -- A tuple with n fields has a constructor and n projections.
 -- This function will also declare the constructor and projection functions.
 mkTupleSort :: Context                         -- ^ Context
```

Comment:
I think that at this point we shouldn't change `mkTupleSort`. I try not to introduce breaking changes unless it's really necessary (or if things get very ugly over time and a major clean up is needed). We could instead create a wrapper for it, `mkTupleType`, that puts everything into a `TupleType` record. This is mainly interesting if we can introduce other helper functions such as `mkTuple :: TupleType -> AST`, or `projTuple :: TupleType -> Symbol -> AST`, or `indexTuple :: TupleType -> Int -> AST`.  I'm not sure we're able to implement `projTuple` like that because we would need a map from `Symbol` to `FunDecl` and I'm not sure we can sensibly instantiate `Ord Symbol`.

Reply:
Okay so we would have
```haskell
data TupleType = TupleType {
   tupleSort :: Sort
   ...
}
mkTupleSort :: ... -> Sort
mkTupleType :: ... -> TupleType
```
That seems like a sensible naming scheme to me :+1:  And `mkTupleType` will call `mkTupleSort`.